ci_framework/plugins/modules/tempest_list_allowed.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2023, Arx Cruz <acruz@redhat.com>
+# Apache License Version 2.0 (see LICENSE)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: tempest_list_allowed
+
+short_description: Parse filtered tests to tempest
+
+description:
+  - Parse filtered tests to be executed by tempest
+
+options:
+  yaml_file:
+    description:
+    - Path to a yaml file containing the tests in openstack-tempest-skiplist
+      format
+    required: True
+    type: str
+  job:
+    description:
+    - Name of the job to be used in the filter.
+    required: False
+    type: str
+  groups:
+    description:
+    - List of groups to be used in the filter. It has more precedence than job
+    required: False
+    type: list
+    elements: str
+  release:
+    description:
+    - Release version
+    type: str
+    default: master
+
+author:
+  - Arx Cruz (@arxcruz)
+
+'''
+
+EXAMPLES = r'''
+- name: Get list of allowed tests
+  tempest_list_allowed:
+    yaml_file: /tmp/allowed.yaml
+    job: tripleo-ci-centos-8-standalone
+    groups:
+     - default
+'''
+
+RETURN = r'''
+allowed_tests:
+  description:
+    - List of tests filtered
+  returned: Always
+  type: list
+  sample: [
+    "tempest.api.volume.test_volumes_snapshots.VolumesSnapshotTestJSON"
+    ]
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+import yaml
+
+
+def _filter_allowed_tests(groups, release, job, yaml_file):
+    tests = []
+    if len(yaml_file) > 0:
+        if groups:
+            everything = yaml_file.get('groups', [])
+            for group in groups:
+                group_tests = list(filter(
+                    lambda x:
+                        x.get('name', '') == group, everything))
+                group_tests = list(filter(
+                    lambda x:
+                        any(r in ['all', release] for r in x.get(
+                            'releases', [])),
+                        group_tests))
+                if len(group_tests) > 0:
+                    tests = tests + group_tests
+        if job:
+            everything = yaml_file.get('jobs', [])
+            job_tests = list(filter(
+                lambda x: x.get('name', '') == job,
+                everything))
+            job_tests = list(filter(
+                lambda x:
+                    any(r in ['all', release] for r in x.get(
+                        'releases', [])),
+                    job_tests))
+            if len(job_tests) > 0:
+                tests = job_tests
```

Comment:
We did all of https://github.com/openstack-k8s-operators/ci-framework/pull/216/files#diff-ba47e19b85bfc2ff06383666f292b5cc6c53c87706c80fc228149d019dafb368R75-R87 but in the end we are returning only the jobs_test. What if we firstly try to do this path about jobs and if job_tests is not empty just return avoid computing the groups_tests.

Reply:
It's ok for me then.